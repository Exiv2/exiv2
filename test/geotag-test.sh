#!/usr/bin/env bash
# Test driver for geotag

source ./functions.source

(   jpg=FurnaceCreekInn.jpg
    gpx=FurnaceCreekInn.gpx
    copyTestFiles $jpg $gpx

    echo --- show GPSInfo tags ---
    runTest                      exiv2 -pa --grep GPSInfo $jpg
    tags=$(runTest               exiv2 -Pk --grep GPSInfo $jpg  | tr -d '\r') # MSVC puts out cr-lf lines
    echo --- deleting the GPSInfo tags
    for tag in $tags; do runTest exiv2 -M"del $tag" $jpg; done
    runTest                      exiv2 -pa --grep GPS     $jpg
    echo --- run geotag ---
    runTest                      geotag -ascii -tz -8:00 $jpg $gpx | cut -d' ' -f 2-
    echo --- show GPSInfo tags ---
    runTest                      exiv2 -pa --grep GPSInfo $jpg
) > $results 2>&1

reportTest

# That's all Folks!
##
