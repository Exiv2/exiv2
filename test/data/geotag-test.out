--- show GPSInfo tags ---
Exif.GPSInfo.GPSVersionID                    Byte        4  2.2.0.0
Exif.GPSInfo.GPSLatitudeRef                  Ascii       2  North
Exif.GPSInfo.GPSLatitude                     Rational    3  36 deg 26' 54"
Exif.GPSInfo.GPSLongitudeRef                 Ascii       2  West
Exif.GPSInfo.GPSLongitude                    Rational    3  116 deg 51' 18"
Exif.GPSInfo.GPSAltitudeRef                  Byte        1  Below sea level
Exif.GPSInfo.GPSAltitude                     Rational    1  14.3 m
Exif.GPSInfo.GPSTimeStamp                    Rational    3  09:54:28
Exif.GPSInfo.GPSMapDatum                     Ascii       7  WGS-84
Exif.GPSInfo.GPSProcessingMethod             Undefined  18  charset=Ascii HYBRID-FIX
Exif.GPSInfo.GPSDateStamp                    Ascii      20  2008:05:08 09:54:28
--- deleting the GPSInfo tags
--- run geotag ---
116deg51'18"W 036deg26'54"N -14.282  -3
--- show GPSInfo tags ---
Exif.GPSInfo.GPSVersionID                    Byte        4  2.2.0.0
Exif.GPSInfo.GPSLatitudeRef                  Ascii       2  North
Exif.GPSInfo.GPSLatitude                     Rational    3  36 deg 26' 54"
Exif.GPSInfo.GPSLongitudeRef                 Ascii       2  West
Exif.GPSInfo.GPSLongitude                    Rational    3  116 deg 51' 18"
Exif.GPSInfo.GPSAltitudeRef                  Byte        1  Below sea level
Exif.GPSInfo.GPSAltitude                     Rational    1  14.3 m
Exif.GPSInfo.GPSTimeStamp                    Rational    3  09:54:28
Exif.GPSInfo.GPSMapDatum                     Ascii       7  WGS-84
Exif.GPSInfo.GPSProcessingMethod             Undefined  18  charset=Ascii HYBRID-FIX
Exif.GPSInfo.GPSDateStamp                    Ascii      20  2008:05:08 09:54:28
